### PR TITLE
Update PKGBUILD (add PETSc and NumPy, run tests, update for v2.1.0)

### LINKED ISSUE #628: AUR package is missing PETSc
Our [AUR package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=precice) does not specify PETSc as a dependency:

```
depends=('openmpi' 'boost')
```

It also builds with `PETSc=off`:
```
cmake . \
	  -DCMAKE_BUILD_TYPE=Release \
	  -DCMAKE_INSTALL_PREFIX=/usr \
          -DBUILD_SHARED_LIBS=ON \
          -DPETSC=off \
          -DPYTHON=off
```
this means that somebody that installs preCICE from AUR would not be able to immediately run an RBF-mapping tutorial.

I am not sure if this was skipped because of compatibility issues with PETSc, but would also give us useful insights in our CI.

Related to #626.

### PR DESCRIPTION
**Pull request by MakisH**
We currently have a `PKGBUILD` for Arch Linux in our repository, which is a copy of the one @floli maintains on AUR. This file is mainly used as a template, but still it helps to keep it up to date.

This PR:
- Adds PETSc, as discussed in #628, which can also be found only on AUR.
- Add a `check()` step, which runs the tests. Users [can disable this using `BUILDENV+=('!check')`](https://wiki.archlinux.org/index.php/Creating_packages). This could help us catch new, dependency-induced errors.
- For the template only, it changes the maintainer information to "The preCICE developers" as more generic.
- Increases the version to v2.1.0. In any case, the current combination of v1.6.1 + CMake variables does not work.
- Fixes the indentation (spaces vs tabs)

I tested this on Manjaro (version 20, up-to-date packages) with the C++ elastictube1d example.

Review requests:
- Try building on any Arch-based system and running your favourite coupled solver.

Closes #628.

### CODE REVIEW

**Comment by MakisH** [comment]
I also added a `check()` step, which runs the tests. Users [can disable this using `BUILDENV+=('!check')`](https://wiki.archlinux.org/index.php/Creating_packages). This could help us catch new, dependency-induced errors.

<details><summary>However, this already revealed a failing test case (`precice.parallel`):</summary>
==> Starting check()...
Running tests...
Test project /var/tmp/pamac-build-tester/precice/src/precice-2.1.0
      Start  1: precice.acceleration
 1/27 Test  #1: precice.acceleration ......................   Passed    0.57 sec
      Start  2: precice.action
 2/27 Test  #2: precice.action ............................   Passed    0.41 sec
      Start  3: precice.com
 3/27 Test  #3: precice.com ...............................   Passed    0.70 sec
      Start  4: precice.cplscheme
 4/27 Test  #4: precice.cplscheme .........................   Passed    0.81 sec
      Start  5: precice.io
 5/27 Test  #5: precice.io ................................   Passed    0.38 sec
      Start  6: precice.m2n
 6/27 Test  #6: precice.m2n ...............................   Passed    0.48 sec
      Start  7: precice.mapping
 7/27 Test  #7: precice.mapping ...........................   Passed    0.39 sec
      Start  8: precice.mapping.petrbf
 8/27 Test  #8: precice.mapping.petrbf ....................   Passed    1.94 sec
      Start  9: precice.math
 9/27 Test  #9: precice.math ..............................   Passed    0.33 sec
      Start 10: precice.mesh
10/27 Test #10: precice.mesh ..............................   Passed    0.39 sec
      Start 11: precice.partition
11/27 Test #11: precice.partition .........................   Passed    0.70 sec
      Start 12: precice.interface
12/27 Test #12: precice.interface .........................   Passed    0.32 sec
      Start 13: precice.serial
13/27 Test #13: precice.serial ............................   Passed    2.59 sec
      Start 14: precice.parallel
14/27 Test #14: precice.parallel ..........................***Timeout  60.08 sec
      Start 15: precice.query
15/27 Test #15: precice.query .............................   Passed    0.37 sec
      Start 16: precice.testing
16/27 Test #16: precice.testing ...........................   Passed    0.42 sec
      Start 17: precice.utils
17/27 Test #17: precice.utils .............................   Passed    0.41 sec
      Start 18: precice.xml
18/27 Test #18: precice.xml ...............................   Passed    0.57 sec
      Start 19: precice.solverdummy.build.cpp
19/27 Test #19: precice.solverdummy.build.cpp .............   Passed    2.52 sec
      Start 20: precice.solverdummy.build.c
20/27 Test #20: precice.solverdummy.build.c ...............   Passed    1.48 sec
      Start 21: precice.solverdummy.build.fortran
21/27 Test #21: precice.solverdummy.build.fortran .........   Passed    2.20 sec
      Start 22: precice.solverdummy.run.cpp-cpp
22/27 Test #22: precice.solverdummy.run.cpp-cpp ...........   Passed    0.29 sec
      Start 23: precice.solverdummy.run.c-c
23/27 Test #23: precice.solverdummy.run.c-c ...............   Passed    0.32 sec
      Start 24: precice.solverdummy.run.fortran-fortran
24/27 Test #24: precice.solverdummy.run.fortran-fortran ...   Passed    0.33 sec
      Start 25: precice.solverdummy.run.cpp-c
25/27 Test #25: precice.solverdummy.run.cpp-c .............   Passed    0.29 sec
      Start 26: precice.solverdummy.run.cpp-fortran
26/27 Test #26: precice.solverdummy.run.cpp-fortran .......   Passed    0.34 sec
      Start 27: precice.solverdummy.run.c-fortran
27/27 Test #27: precice.solverdummy.run.c-fortran .........   Passed    0.34 sec

96% tests passed, 1 tests failed out of 27

Label Time Summary:
Solverdummy    =   8.12 sec*proc (9 tests)
petsc          =   1.94 sec*proc (1 test)

Total Test time (real) =  80.02 sec

The following tests FAILED:
	 14 - precice.parallel (Timeout)
Errors while running CTest
make: *** [Makefile:150: test] Error 8
==> ERROR: A failure occurred in check().
    Aborting...
</details>

**Comment by floli** [comment]
Just let you know, that I will take this script over to the AUR, as soon as this PR is merged.

**Comment by MakisH** [comment]
I am not sure why, but the tests do not fail anymore. Probably some problem with my fresh installation or my VM.

The PythonActions tests fail, which is why I still have them off.

**Comment by MakisH** [comment]
- I also added `python-numpy` to the dependencies, the tests seem to pass now (no clue why they were failing locally before).
- I also added the `-DPRECICE_MPICommunication=ON` to make it clear how to disable the feature.

I considered adding `python-numpy` and `petsc` to `optdepends()`, but I think this will only cause confusion at the moment. The user will not always (if at all) be asked to install it and, if choosing to not install it, CMake will simply complain. We can better manage the optional dependencies in the future, maybe also inside CMake.

**Comment by MakisH** [comment]
I checked the recipe with [namcap](https://wiki.archlinux.org/index.php/Namcap). I fixed the following:

- `PKGBUILD (precice) W: Non-unique source name (v2.1.0.tar.gz). Use a unique filename.` --> prefixed the source filename with `$pkgname-$pkgver.tar.gz::`
- `PKGBUILD (precice) I: Missing Contributor tag` --> added a `# Contributor: The preCICE developers <info@precice.org>`. I am not sure what the difference between `contributor` and `maintainer` is, but feel free to adjust before uploading, if needed.

**Comment by floli** [comment]
>     * `PKGBUILD (precice) I: Missing Contributor tag` --> added a `# Contributor: The preCICE developers <info@precice.org>`. I am not sure what the difference between `contributor` and `maintainer` is, but feel free to adjust before uploading, if needed.

Maintainer is the current maintainer of the PKGBUILD, that is, me. Contributer are previous maintainer. As long as I own the AUR package, please leave the maintainer as it is and no contributor.

If precice somewhat officially want to take over maintainership of the AUR package, I am happy to turn it over.

**Comment by MakisH** [comment]
@floli No problem with any of the options. Since you are currently up for maintaining the package, I added you again as the maintainer and removed the contributor tag.

File: tools/releasing/packaging/Arch_User_Repository/PKGBUILD
Comment (MakisH) [reply tools/releasing/packaging/Arch_User_Repository/PKGBUILD]: Good point, I included it after boost to keep the necessary dependencies first.

**Comment by MakisH** [comment]
@fsimonis Do we really want to use `-DCMAKE_BUILD_TYPE='None'`? Is there anything else I should take into account or should I go through the process of comparing the builds?

There is also the option to explicitly set `-O2` in preCICE, which could make the builds more reproducible, but maybe also confuse people.

**Comment by MakisH** [comment]
Regarding the build type, I built both with verbose make output, here are the complete files:
[None.log](https://github.com/precice/precice/files/5023001/None.log)
[Release.log](https://github.com/precice/precice/files/5023002/Release.log)

It seems to me that our `Release` only adds `-O3 -DNDEBUG`:

```diff
- /usr/bin/c++  -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_ASIO_ENABLE_OLD_SERVICES -DBOOST_ATOMIC_DYN_LINK -DBOOST_CHRONO_DYN_LINK -DBOOST_DATE_TIME_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_GEOMETRY_DISABLE_DEPRECATED_03_WARNING -DBOOST_LOG_DYN_LINK -DBOOST_LOG_SETUP_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SYSTEM_DYN_LINK -DBOOST_THREAD_DYN_LINK -DBOOST_UNIT_TEST_FRAMEWORK_DYN_LINK -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -Dprecice_EXPORTS -I/usr/include/libxml2 -I/opt/petsc/linux-c-opt/include -I/usr/include/python3.8 -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/extras/bindings/c/include -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/extras/bindings/fortran/include -isystem /usr/include/eigen3 -isystem /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/thirdparty/prettyprint/include -isystem /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/thirdparty/json/include -isystem /usr/lib/python3.8/site-packages/numpy/core/include  -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -O3 -DNDEBUG -fPIC   -pthread -std=c++11 -o CMakeFiles/precice.dir/src/acceleration/Acceleration.cpp.o -c /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src/acceleration/Acceleration.cpp
+ /usr/bin/c++  -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_ASIO_ENABLE_OLD_SERVICES -DBOOST_ATOMIC_DYN_LINK -DBOOST_CHRONO_DYN_LINK -DBOOST_DATE_TIME_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_GEOMETRY_DISABLE_DEPRECATED_03_WARNING -DBOOST_LOG_DYN_LINK -DBOOST_LOG_SETUP_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SYSTEM_DYN_LINK -DBOOST_THREAD_DYN_LINK -DBOOST_UNIT_TEST_FRAMEWORK_DYN_LINK -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -Dprecice_EXPORTS -I/usr/include/libxml2 -I/opt/petsc/linux-c-opt/include -I/usr/include/python3.8 -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/extras/bindings/c/include -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/extras/bindings/fortran/include -isystem /usr/include/eigen3 -isystem /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/thirdparty/prettyprint/include -isystem /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/thirdparty/json/include -isystem /usr/lib/python3.8/site-packages/numpy/core/include  -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC   -pthread -std=c++11 -o CMakeFiles/precice.dir/src/acceleration/Acceleration.cpp.o -c /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src/acceleration/Acceleration.cpp
``` 

Interestingly, `None` also hides a few developer-oriented warnings:
```
[ 39%] Building CXX object CMakeFiles/precice.dir/src/math/barycenter.cpp.o
/usr/bin/c++  -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_ASIO_ENABLE_OLD_SERVICES -DBOOST_ATOMIC_DYN_LINK -DBOOST_CHRONO_DYN_LINK -DBOOST_DATE_TIME_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_GEOMETRY_DISABLE_DEPRECATED_03_WARNING -DBOOST_LOG_DYN_LINK -DBOOST_LOG_SETUP_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SYSTEM_DYN_LINK -DBOOST_THREAD_DYN_LINK -DBOOST_UNIT_TEST_FRAMEWORK_DYN_LINK -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -Dprecice_EXPORTS -I/usr/include/libxml2 -I/opt/petsc/linux-c-opt/include -I/usr/include/python3.8 -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/extras/bindings/c/include -I/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/extras/bindings/fortran/include -isystem /usr/include/eigen3 -isystem /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/thirdparty/prettyprint/include -isystem /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/thirdparty/json/include -isystem /usr/lib/python3.8/site-packages/numpy/core/include  -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -O3 -DNDEBUG -fPIC   -pthread -std=c++11 -o CMakeFiles/precice.dir/src/math/barycenter.cpp.o -c /var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src/math/barycenter.cpp
/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src/mapping/config/MappingConfiguration.cpp: In member function ‘precice::mapping::MappingConfiguration::Timing precice::mapping::MappingConfiguration::getTiming(const string&) const’:
/var/tmp/pamac-build-tester/precice/src/precice-2.1.0/src/mapping/config/MappingConfiguration.cpp:415:1: warning: control reaches end of non-void function [-Wreturn-type]
  415 | }
      | ^
```

**Comment by MakisH** [comment]
@fsimonis I applied all the appropriate changes to comply with the CMake-related suggestions:

- The build type is now `None` instead of `Release`.
- I manually set `-DCMAKE_CXX_FLAGS="-O2 -DNDEBUG"` to keep the optimization to `O2` and to also disable assertions.
- I set `-Wno-dev` to disable CMake warnings intended for developers.

What I did not change:
- We already set `-DCMAKE_INSTALL_PREFIX=/usr`
- I did not use the newer CMake way of definings the build directory (no reason to)
- I did not set `CMAKE_SKIP_INSTALL_RPATH=YES` (Namcap did not complain)

Ready for reviewing again.

File: tools/releasing/packaging/Arch_User_Repository/PKGBUILD
Comment (fsimonis) [inline tools/releasing/packaging/Arch_User_Repository/PKGBUILD:26]:
```suggestion
```

Omitting the flags will detect the flags defined in the `/etc/makepkg.conf`.

**Review by fsimonis — APPROVED** [review]
Everything works as expected apart from the flags.

**Comment by MakisH** [comment]
Travis fails with known unrelated issues. In any case, Travis does not check the PKGBUILD. Me and @fsimonis have checked it, so I squash & merge.

@floli  ready to upload to AUR. Keep an eye on updating the package soon for a v2.1.1.

**Pull request merged**
